### PR TITLE
Creating a single page for every book

### PR DESCRIPTION
**Pull request by AhmadAlkaati**
* Creating a single page of details for every book and making it responsive.
* Creating a page for 'read more' button that exists in main page and making it responsive.

### CODE REVIEW

File: app.js [hunk 1]
```diff
@@ -1,120 +1,28 @@
...
-      bookDiv += `<div class='book'> ${title} ${author} </div> `;
-      console.log(title);
-      console.log(author);
-      console.log(bookDiv);
+  fetch("https://bookshop-api.mirkwood.dev/books/")
+    .then(function (re) {
+      return re.json();
+    })
+    .then(function (data) {
+      return displayBooks(data);
```
Comment (josipgrgic) [inline app.js]: `displayBooks` returns no data to the caller so the return keyword is not needed

File: books/book.js [hunk 2]
```diff
@@ -0,0 +1,40 @@
+let currentUrl = new URL(location.href); // creating new url that contains current url
+let searchParams = new URLSearchParams(currentUrl.search); //  searching in current  url params
+let currentUrlId = searchParams.get("id"); //  getting current url id// Fetching book when page is loaded
```
Comment (josipgrgic) [inline books/book.js]: A better name for this variable would be just `bookId`

File: books/book.js [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+  let title = ` <h2 class = 'title' > ${book.title} </h2>`;
+  let author = `<h5 class = 'author'> By (author)  ${book.author.first_name} ${book.author.last_name}</h5> `;
+  let bookYear = `<h5 class = 'book-year'> Originally published: ${book.year} </h5>`;
+  let topic = `<h5 class = 'topic'> Topic: ${book.topics[0].name} </h5>`;
+  let synopsis = `<p class = 'synopsis'> ${book.synopsis} </p> `;
+  let bookPrice = `<h2> ${book.price} € </h2>`;
+  let bookPurchaseH3 = `<h3> Free delivery worldwide</h3>`;
+  let bookPurchaseP = `  <p> <i class="far fa-check-circle"></i> Available. Dispatched from Australia in 3  business  days. </p>`;
+  let buttonBasket = `<button class = 'basket' > Add to basket </button>`;
+  let buttonWish = `<button class = 'wishlist' > Add to wishlist </button>`;
```
Comment (josipgrgic) [inline books/book.js]: Since we will buy a single book from the details page, you could add another "Buy now" button. When this button is clicked, a call to the REST API purchase endpoint should be made, but that's another chunk of work.

For now just add the button :)

File: books/book.html [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+  <body>
+    <header>
+      <div class="container">
+        <h1><a href="../index.html"> Books for all!</a></h1>
+      </div>
+    </header>
+    <main class="main-content"></main>
+
+    <footer>
+      <h5>&copy; 2020 Books For All, All Rights Reserved</h5>
```
Comment (josipgrgic) [inline books/book.html:27]: Try to fix this footer to the bottom of the page so it doesn't float in the middle of the page. This and other styling work can also be done in another PR if you have some more plans with this page

**Review by josipgrgic — CHANGES_REQUESTED** [review]
Good job in general. The code is tidy and well-formatted

The about page could easily be another PR so we keep this one only focused on the book detail page, but it's fine as is.

Please resolve the comments I made after which you can merge this branch to main

File: app.js [hunk 5]
```diff
@@ -1,120 +1,28 @@
...
+    })
+    .catch(function (err) {
+      console.log(err);
     });
+}
+
+function displayBooks(data) {
+  let bookDiv = "";
+  data.forEach((book) => {
+    let title = `<a class = 'title' href= "books/book.html?id=${book.id}">  ${book.title}</a>  `;
```
Comment (josipgrgic) [inline app.js]: Variables you never reassign later in code should be declared with `const `instead of with `let`. You can read more about it here: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/let

File: app.js (hunk 1 above)
Comment (AhmadAlkaati) [reply app.js]: This is fixed

File: books/book.js (hunk 2 above)
Comment (AhmadAlkaati) [reply books/book.js]: This is fixed as well:)

File: books/book.js (hunk 3 above)
Comment (AhmadAlkaati) [reply books/book.js]: Done this

File: books/book.html (hunk 4 above)
Comment (AhmadAlkaati) [reply books/book.html:27]: I fixed the footer but can you tell me when does it float in middle of the page ? at which screen's size?

File: app.js (hunk 5 above)
Comment (AhmadAlkaati) [reply app.js]: Changed all variables that are used once only to const

**Comment by AhmadAlkaati** [comment]
@josipgrgic  Hi Josip , once more I would like to thank you for taking the time to review my code :) I did the things you talked about

File: books/book.html (hunk 4 above)
Comment (josipgrgic) [reply books/book.html:27]: image
 Here is a screenshot

Comment (AhmadAlkaati) [reply books/book.html:27]: @josipgrgic  I think its fixed now

Comment (AhmadAlkaati) [reply books/book.html:27]: Capture

**Review by josipgrgic — APPROVED** [review]
